### PR TITLE
📊 hmd: add data

### PR DESCRIPTION
**Pull request by lucasrodes**
Adds new HMD dataset (up to Grapher) and HLD (only Snapshot)

### HMD
- [x] `snapshots/hmd/2023-09-18/hmd.*`
- [x] `etl/steps/data/meadow/hmd/2023-09-19/hmd.py`
- [x] `etl/steps/data/garden/hmd/2023-09-19/hmd.*`
- [x] `etl/steps/data/grapher/hmd/2023-09-19/hmd.py`

### HLD
- [x] `snapshots/hmd/2023-09-19/hld.*`

### Others
Other files were updated to update with latest ETL requirements (e.g. set indices). This does not need careful revision.

- [x] `etl/steps/data/meadow/hmd/2022-11-04/life_tables.py`
- [x] `etl/steps/data/garden/hmd/2022-11-04/life_tables.py`
- [x] `etl/steps/data/meadow/excess_mortality/latest/*`
- [x] `etl/steps/data/garden/excess_mortality/latest/*`

### CODE REVIEW

File: etl/steps/data/meadow/hmd/2023-09-19/hmd.py [hunk 1]
```diff
@@ -0,0 +1,267 @@
...
+    return tb
+
+
+def proces_missing_data_lt(tb: Table) -> Table:
+    """Check and process missing data.
+
+    Missing data comes as '.', instead replace these with NaNs.
+    """
+    # Expected missing data
+    frac_rows_missing_expected = 0.01
```
Comment (spoonerf) [inline etl/steps/data/meadow/hmd/2023-09-19/hmd.py:224]: Maybe good to set these hard-coded values at the top of the file so it's easier to revisit them when running an update?

File: etl/steps/data/garden/hmd/2023-09-19/hmd.meta.yml [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+        short_unit: years
+        unit: years
+        description_short: Period life expectancy at exact age x (in years).
+      exposure:
+        title: Exposure-to-risk
+        unit: person-years
+        description_short: Total person-years lived in the age interval [x, x + 1) during calendar year t.
+        description_from_producer: |-
+          Estimates of the population exposed to the risk of death during some agetime interval are based on annual (January 1st) population estimates, with small corrections that reflect the timing of deaths during the interval. Period exposure estimations are based on assumptions of uniformity in the distribution of events except when historical monthly birth data are available.
+      # testing_variable:
```
Comment (spoonerf) [inline etl/steps/data/garden/hmd/2023-09-19/hmd.meta.yml:56]: Maybe below here could be deleted?

File: snapshots/hmd/2023-09-19/hld.zip.dvc [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    url_main: https://www.lifetable.de/
+    url_download: https://www.lifetable.de/File/GetDocument/data/hld.zip
+    date_accessed: 2023-09-19
+
+    # License
+    license:
+      url: https://www.lifetable.de/Data/CitationGuidelines
+      name: CC BY 4.0
+
+  # License (same as origin.license, for backwards compatibility)
```
Comment (spoonerf) [inline snapshots/hmd/2023-09-19/hld.zip.dvc:32]: Is this license needed - the schema I have doesn't seem to like it

File: etl/steps/data/garden/hmd/2023-09-19/hmd.meta.yml [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+# http://docs.owid.io/projects/etl/en/latest/architecture/metadata/reference/tables/
+tables:
+  hmd:
+    variables:
+      central_death_rate:
+        title: Central death rate
+        description_short: Central death rate between ages x and x+n.
+        unit: deaths per people
+      probability_of_death:
+        title: Probability of death
```
Comment (spoonerf) [inline etl/steps/data/garden/hmd/2023-09-19/hmd.meta.yml:20]: Suggest multiplying these values by 100 and having this as a % - currently the values are so small they don't display well on grapher.

File: etl/steps/data/garden/hmd/2023-09-19/hmd.meta.yml [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+
+# Learn more about the available fields:
+# http://docs.owid.io/projects/etl/en/latest/architecture/metadata/reference/tables/
+tables:
+  hmd:
+    variables:
+      central_death_rate:
+        title: Central death rate
+        description_short: Central death rate between ages x and x+n.
+        unit: deaths per people
```
Comment (spoonerf) [inline etl/steps/data/garden/hmd/2023-09-19/hmd.meta.yml:18]: Not sure what this unit means exactly

Comment (spoonerf) [reply etl/steps/data/garden/hmd/2023-09-19/hmd.meta.yml:18]: Is it the number of deaths per capita?

**Review by spoonerf — APPROVED** [review]
Looks good - just a few minor suggestions

File: snapshots/hmd/2023-09-18/hmd.zip.dvc [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+    url_main: https://www.mortality.org/Data/ZippedDataFiles
+    date_accessed: 2023-09-18
+
+    # License
+    license:
+      url: https://www.mortality.org/Data/UserAgreement
+      name: CC BY 4.0
+
+  # License (same as origin.license, for backwards compatibility)
+  license:
```
[the remark this answers is not in the document]
Comment (lucasrodes) [reply snapshots/hmd/2023-09-18/hmd.zip.dvc:84]: Not anymore, thanks! Removing

File: etl/steps/data/garden/hmd/2023-09-19/hmd.meta.yml (hunk 4 above)
Comment (lucasrodes) [reply etl/steps/data/garden/hmd/2023-09-19/hmd.meta.yml:20]: Great suggestion. I'll look into this in a future PR, might change what is displayed in the grapher dataset.

**Pull request merged**